apis/iam/v1beta1/serviceaccount_types.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// ServiceAccountParameters defines parameters for a desired IAM ServiceAccount
+// https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts
+type ServiceAccountParameters struct {
+	// Required. The resource name of the project associated with the service
+	// accounts, such as `projects/my-project-123`.
+	// note that this is only used for the initial creation request, eg `Name` field in
+	// https://godoc.org/google.golang.org/genproto/googleapis/iam/admin/v1#CreateServiceAccountRequest
+	// +immutable
+	//ProjectName string `json:"name,omitempty"`
+
+	// Required. The account id that is used to generate the service account
+	// email address and a stable unique id. It is unique within a project,
+	// must be 6-30 characters long, and match the regular expression
+	// `[a-z]([-a-z0-9]*[a-z0-9])` to comply with RFC1035.
+	// +immutable
+	AccountID string `json:"accountId,omitempty"`
```

Comment:
I think this should be represented by the [External Name annotation](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#external-resource-labeling).

Reply:
You were 100% correct!